### PR TITLE
Add methods for graph subranges (needed for TC)

### PR DESCRIPTION
**Pull request by pingle14**
# Description
Add localVertices and edges with windowsz and offset so can get subranges of the vertices and edges

### CODE REVIEW

File: include/pando-lib-galois/graphs/dist_local_csr.hpp [hunk 1]
```diff
@@ -449,11 +449,34 @@ class DistLocalCSR {
...
+
+    auto beg = v.edgeBegin + offset_st;
+    if (beg > v1.edgeBegin)
+      return RefSpan<galois::HalfEdge>(v.edgeBegin, 0);
+    if (beg + window_sz <= v1.edgeBegin)
+      return RefSpan<galois::HalfEdge>(beg, window_sz);
+
+    return RefSpan<galois::HalfEdge>(beg, v1.edgeBegin - beg);
+  }
+
```
Comment (AdityaAtulTewari) [inline include/pando-lib-galois/graphs/dist_local_csr.hpp:479]: Why do you need to do this? Is it not enough to have doAll infer the locality? Or to just operate on the local csr?

**Review by AdityaAtulTewari — COMMENTED** [review]
Please explain this.

File: include/pando-lib-galois/graphs/dist_local_csr.hpp (hunk 1 above)
Comment (pingle14) [reply include/pando-lib-galois/graphs/dist_local_csr.hpp]: - `EdgeRange edges` allows me to chunk the edges for a particular vertex and use that subrange in my `doAll`. This is necessary for chunked TC, as I can execute TC on a subset of outgoing edges each round.
- However, I do see that `localVertices` method is not required as we have a `getLocalCSR.vertices()` I could call instead. I remove `localVertices` in `https://github.com/utcs-scea/pando-mono/pull/31/commits/edb8320623412b48d7747d52b4d43f68638b9e83`

Comment (AdityaAtulTewari) [reply include/pando-lib-galois/graphs/dist_local_csr.hpp]: okay make sure that works before changing this. But I would prefer that.

Comment (pingle14) [reply include/pando-lib-galois/graphs/dist_local_csr.hpp]: `getLocalCSR.vertices()` works, but for chunking vertices and edges, I added methods to grab a subset of vertices and edges.

**Review by AdityaAtulTewari — APPROVED** [review]
Lgtm.

**Pull request merged**